x-pack/plugins/security/server/lib/audit_logger.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class SecurityAuditLogger {
+  constructor(config, auditLogger) {
+    this._enabled = config.get('xpack.security.audit.enabled');
+    this._auditLogger = auditLogger;
+  }
+
+  savedObjectsAuthorizationFailure(username, action, types, missing) {
```

Comment:
Is `args` intentionally omitted from the failure case?

Reply:
It was intentional, but it might not be what we really want. My reasoning was that in the case where we're denying the user from performing an action, the specifics of that action aren't relevant if they weren't able to perform it. So we're only logging the information that we used to deny the request.

Reply:
That's a fair argument. My counter-argument is:
1. Adding `args` will make parsing the failure entries more consistent with the `success` entries
2. For investigating incidents, it could be useful to know what saved objects (or request payloads) the user was attempting to send, malicious or otherwise.

I don't have a strong opinion either way here, so I'm fine with whatever you decide. This is something that's easily added later on if we want it.